reconcile/gitlab_housekeeping.py

```diff
@@ -19,6 +19,26 @@
 
 QONTRACT_INTEGRATION = 'gitlab-housekeeping'
 
+def clean_pipelines(dry_run, gl, pipeline_timeout=60):
+    DATE_FORMAT = '%Y-%m-%dT%H:%M:%S.%fZ'
+    now = datetime.utcnow()
+
+    pipelines = gl.get_pipelines(status='pending')
+
+    for p in pipelines:
+        last_update_date = datetime.strptime(p.updated_at, DATE_FORMAT)
```

Comment:
What is the data type of this `p`? I'd suggest converting it to seconds since epoch and just do normal integer operations. The `datetime` stuff is great if you have to print stuff to humans, or do calculations of the form "the last Monday of next month, two years from now". But for the duration of a computer operation, stick to `time.time`.